cromwell-backend/src/main/scala/cromwell/backend/config/BackendConfiguration.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.backend.config
+
+import com.typesafe.config.ConfigFactory
+
+import scala.collection.JavaConversions._
+
+/**
+  * Defines a backend configuration.
+  * @param name Backend name.
+  * @param initClass Initialization class for the specific backend.
+  */
+case class BackendConfigurationEntry(name: String, initClass: String)
+
+object BackendConfiguration {
+  val config = ConfigFactory.load()
+  val backendCfg = config.getConfig("backend")
+  val defaultBackend = backendCfg.getString("default")
+  val backendProviders = backendCfg.getConfigList("providers")
+  val backendList = backendProviders.map(entry =>
+    BackendConfigurationEntry(entry.getString("name"), entry.getString("initClass"))).toList
+
+  def apply(): BackendConfiguration = new BackendConfiguration(backendList, defaultBackend)
+}
+
+/**
+  * Retrieves backend configuration.
+  * @param backendList List of backend entries in configuration file.
+  * @param defaultBackend Backend name to be used as default.
+  */
+class BackendConfiguration(backendList: List[BackendConfigurationEntry], defaultBackend: String) {
```

Comment:
@francares Did you intend to accept _only_ a `scala.List` of backends, and also only accept/return `scala.List` elsewhere?